src/index.js

```diff
@@ -1,17 +1,24 @@
 export default function (Alpine) {
   Alpine.directive(
-    "count",
-    (el, { expression, modifiers }, { evaluateLater, effect }) => {
-      let maxLength = modifiers[0] || false;
-      let getInputValue = evaluateLater(expression);
+    'count',
+    (el, { expression, modifiers }, { evaluateLater, effect, Alpine }) => {
+      let firstModifier = modifiers[0]
+      let modifierNumber = Number(firstModifier)
+      let modifierTarget =
+        !modifierNumber && document.querySelector(`[x-ref="${firstModifier}"]`)
```

Comment:
Im not sure it possible, but maybe this could be `$refs[firstModifier]` 
Source: https://alpinejs.dev/magics/refs

Reply:
You can't access `$refs` in custom directives.